tests/src/test/java/com/cloudera/director/google/sql/GoogleCloudSQLInstanceTemplateConfigurationValidatorTest.java

```diff
@@ -0,0 +1,251 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.cloudera.director.google.sql;
+
+import static com.cloudera.director.google.sql.GoogleCloudSQLInstanceTemplateConfigurationProperty.TIER;
+import static com.cloudera.director.google.sql.GoogleCloudSQLInstanceTemplateConfigurationValidator.INVALID_PREFIX_LENGTH_MSG;
+import static com.cloudera.director.google.sql.GoogleCloudSQLInstanceTemplateConfigurationValidator.INVALID_PREFIX_MSG;
+import static com.cloudera.director.google.sql.GoogleCloudSQLInstanceTemplateConfigurationValidator.INVALID_TIER_MSG;
+import static com.cloudera.director.google.sql.GoogleCloudSQLInstanceTemplateConfigurationValidator.PREFIX_MISSING_MSG;
+import static com.cloudera.director.google.sql.GoogleCloudSQLProviderConfigurationProperty.REGION_SQL;
+
+import static com.cloudera.director.spi.v1.model.InstanceTemplate.InstanceTemplateConfigurationPropertyToken.INSTANCE_NAME_PREFIX;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Matchers.eq;
+import static org.mockito.Mockito.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import com.cloudera.director.google.TestUtils;
+import com.cloudera.director.google.internal.GoogleCredentials;
+import com.cloudera.director.google.shaded.com.google.api.client.googleapis.json.GoogleJsonResponseException;
+import com.cloudera.director.google.shaded.com.google.api.client.googleapis.testing.json.GoogleJsonResponseExceptionFactoryTesting;
+import com.cloudera.director.google.shaded.com.google.api.client.testing.json.MockJsonFactory;
+import com.cloudera.director.google.shaded.com.google.api.services.sqladmin.SQLAdmin;
+import com.cloudera.director.spi.v1.model.ConfigurationPropertyToken;
+import com.cloudera.director.spi.v1.model.Configured;
+import com.cloudera.director.spi.v1.model.LocalizationContext;
+import com.cloudera.director.spi.v1.model.exception.PluginExceptionCondition;
+import com.cloudera.director.spi.v1.model.exception.PluginExceptionConditionAccumulator;
+import com.cloudera.director.spi.v1.model.util.DefaultLocalizationContext;
+import com.cloudera.director.spi.v1.model.util.SimpleConfiguration;
+import com.google.common.base.Optional;
+import org.assertj.core.util.Maps;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.util.Collection;
+import java.util.Locale;
+import java.util.Map;
+
+/**
+ * Tests {@link GoogleCloudSQLInstanceTemplateConfigurationValidator}.
+ */
+public class GoogleCloudSQLInstanceTemplateConfigurationValidatorTest {
+
+  private static final String PROJECT_ID = "some-project";
+  private static final String REGION_NAME_1 = "us-central";
```

Comment:
This is not a valid region name. A valid region name has a number, e.g., `us-central1`.